src/transports/tcp.rs

```diff
@@ -88,26 +88,30 @@ impl Default for TcpConnectionConfig {
 pub struct TcpEndpoint {
     pub config: TcpConnectionConfig,
     pub address: SocketAddr,
-    pub stream: TcpStream,
+    pub stream: Limiter<TcpStream>,
```

Comment:
maybe  call it `stream_limiter` to avoid `endpoint.stream.stream` call which is not very "elegant"